### DISCUSSION TITLE
Unable to subcribe to the transient state.

### DISCUSSION THREAD

**Discussion by afkcodes** [pmndrs/zustand#777] [Q&A]
```
const Home = () => {
  const focusRef = useRef(useStore.getState().isSearchBarFocussed);
  useEffect(
    () => useStore.subscribe((state) => (focusRef.current = state.isSearchBarFocussed)),
    []
  );

  console.log(focusRef.current);
  const { isLoading, data } = useQuery('home', getData);
  let cleanedHomeData: CleanedHomeData[] = [];
  const homeData = data?.data as HomeData;

  if (!isLoading && data) {
    cleanedHomeData = objToSortedArr(homeData);
  }
  if (isLoading) {
    return <Loading />;
  }

  if (homeData && homeData.status === 'error') {
    return <Error />;
  }

  return (
    <React.Fragment>
      <FlatList
        data={Object.keys(cleanedHomeData)}
        renderItem={({ index }) => {
          return (
            <Main index={index}>
              {index === 0 ? <SearchBar /> : <></>}
              <MbText>{focusRef.current.toString()}</MbText>
              <MbRadioRenderer item={cleanedHomeData[index] as CleanedHomeDataEntity0} />
            </Main>
          );
        }}
        keyExtractor={() => {
          return (
            new Date().getTime().toString() +
            Math.floor(Math.random() * Math.floor(new Date().getTime())).toString()
          );
        }}
        maxToRenderPerBatch={Math.ceil(Dimensions.get('screen').height / 140)}
        initialNumToRender={Math.ceil(Dimensions.get('screen').height / 140)}
        showsVerticalScrollIndicator={false}
      />
    </React.Fragment>
  );
};

export default Home;
```

I am updating the focus state from the SearchBar component. it works fine if. i use the normal way of getting the state.
Please tell me if i am doing wrong.

Project: React Native 0.67

**Comment by dai-shi** [comment]
How it's different from the expected behavior?

The code looks good. You want to try something like this to see if it's working:
```js
  useEffect(
    () => useStore.subscribe((state) => {
      console.log(state)
      focusRef.current = state.isSearchBarFocussed
    }),
    []
  )
```

**Reply by afkcodes** [reply]
sure trying now.

**Reply by afkcodes** [reply]
I see here the state is coming fine @dai-shi .

**Reply by dai-shi** [reply]
so, what's the issue? i guess that's how "transient updating" works.

**Reply by afkcodes** [reply]
The ref does not reflect the updated value if I log it before re-render I need to use that to show something on the screen. so i think there is no way of doing it without rerendering.

**Reply by dai-shi** [reply]
Do you mean you see `console.log(state)` correctly, but `focusRef.current = state.isSearchBarFocussed` isn't effective? That couldn't happen. Or, there would be a bug somewhere not related with zustand.

**Reply by afkcodes** [reply]
i will try with a minimal reproducer and check as in my project it doesn't seems like it at first glance. Thanks for such a wonderful way of managing state.

**Comment by afkcodes** [comment]
```
const Home = () => {
  const focusRef = useRef(useStore.getState().isSearchBarFocussed);
  useEffect(
    () =>
      useStore.subscribe((state) => {
        console.log(state.isSearchBarFocussed);
        focusRef.current = state.isSearchBarFocussed;
      }),
    []
  );

  return (
    <React.Fragment>
      <Main index={0}>
        <SearchBar />
        <MbText>{focusRef.current.toString()}</MbText>
      </Main>
    </React.Fragment>
  );
};
```

I have removed all the other chunks of code and tried only with transient state logic but still I am getting the state logged correctly from the useEffect but not the ref,

**Reply by dai-shi** [reply]
It sounds to me like an expected behavior. what's your expected behavior?

**Reply by MetaMmodern** [reply]
@dai-shi idk why this thread is closed, I also expirience this issue.
In my case, I have a MUI slider and a store. I wanted to attach the slider to the store value, however on every change I have some heavy computation, thus if I attach directly it lags. 

So I tried transient updates. I attached ref.current to slider value and I see that if I have bare component, it actually updates the store, but the ref stays the same.

So what OP means, is that store updates correctly, everything is fine. ~~But the ref stays the same.~~ Actually even ref gets updated. It's the react that doesn't update UI when ref value updates.


UPDATE: I think now that I've written it it makes sense. a Ref is a mutable thing. React doesn't know when to update stuff if you **mutate**. Thus, my slider position won't change. As well as whatever you're trying to see in UI @afkcodes . So this  is like a fundamental react thing.
